aws/data_source_aws_servicecatalog_constraint.go

```diff
@@ -0,0 +1,96 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	tfservicecatalog "github.com/terraform-providers/terraform-provider-aws/aws/internal/service/servicecatalog"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/servicecatalog/waiter"
+)
+
+func dataSourceAwsServiceCatalogConstraint() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsServiceCatalogConstraintRead,
+
+		Schema: map[string]*schema.Schema{
+			"accept_language": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "en",
+				ValidateFunc: validation.StringInSlice(tfservicecatalog.AcceptLanguage_Values(), false),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"owner": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"parameters": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"portfolio_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"product_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsServiceCatalogConstraintRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+
+	output, err := waiter.ConstraintReady(conn, d.Get("accept_language").(string), d.Get("id").(string))
+
+	if err != nil {
+		return fmt.Errorf("error describing Service Catalog Constraint: %w", err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("error getting Service Catalog Constraint: empty response")
+	}
+
+	acceptLanguage := d.Get("accept_language").(string)
+
+	if acceptLanguage == "" {
+		acceptLanguage = "en"
+	}
+
+	d.Set("accept_language", acceptLanguage)
+
+	d.Set("parameters", output.ConstraintParameters)
+	d.Set("status", output.Status)
+
+	detail := output.ConstraintDetail
```

Comment:
just in case, may want to `nil` check this field before setting the nested fields below